### PR TITLE
Friending

### PR DESCRIPTION
**Pull request by derekdowling**
This brings the friending process up to snuff with pending, requests, following, friends fields.

### CODE REVIEW

**Comment by derekdowling** [comment]
Doesn't yet cover deserializing author responses. That should be handled in #160.

**Comment by derekdowling** [comment]
@KarimBaaba this is good to go. Two failures on integration stuff but that's it and unrelated.

File: api/views/author.py [hunk 1]
```diff
@@ -151,19 +136,12 @@ def retrieve(self, request, pk=None, author_pk=None):
 
             # Can handle this in a permission
             if request.user != author.user:
+                print "Trying to spoof author"
```
Comment (xkb) [inline api/views/author.py:139]: remove

Comment (derekdowling) [inline api/views/author.py:139]: This one is intentional. It's a pretty gross bug to have to find. #GhettoLogging

File: api/views/author.py [hunk 2]
```diff
@@ -225,53 +199,38 @@ def create(self, request, *args, **kwargs):
...
-        if request.get_host() is not settings.FRONTEND_HOST:
-            self.add_request(author, friend)
-            # TODO: check if friend request has been sent already, convert to
-                # friends
-                # self.add_friend(friend, author)
-                # self.add_friend(author, friend)
-                # headers = self.get_success_headers(serializer.data)
-                # return Response(serializer.data, status=status.HTTP_202_ACCEPTED,
-                                # headers=headers)
+        if request.get_host() not in [settings.HOST, settings.FRONTEND_HOST, "testserver"]:
```
Comment (xkb) [inline api/views/author.py:202]: Should I even ask?

Comment (derekdowling) [inline api/views/author.py:202]: Nope, huge hack. Essentially django and the TestClient makes some assumptions about host headers and after trying to fix it for 20 minutes this is what I resorted to. Unfortunately we're using the host header to figure out where the F the api calls are coming from.

File: api/views/author.py [hunk 3]
```diff
@@ -225,53 +199,38 @@ def create(self, request, *args, **kwargs):
...
+                if author.is_pending_friend(friend):
+                    author.add_friend(friend)
+                    friend.add_friend(author)
+                else:
+                    author.add_pending(friend)
+                    friend.add_request(author)
 
         headers = self.get_success_headers(serializer.data)
-        return Response(serializer.data, status=status.HTTP_201_CREATED,
+        return Response(serializer.data, status=status.HTTP_200_OK,
```
Comment (xkb) [inline api/views/author.py]: Is this returning what I think it's returning? Same data as the request?

Comment (derekdowling) [inline api/views/author.py]: I think it's returning a group of friends. I dunno, that's how it was before. I don't want to break any more tests haha.

**Comment by xkb** [comment]
:+1:

File: api/views/author.py (hunk 3 above)
Comment (mraypold) [inline api/views/author.py]: Changing it won't break the tests. It's not tested, as I didn't expect that would be the final response.

**Pull request merged**